vowpalwabbit/parse_regressor.cc

```diff
@@ -349,8 +348,18 @@ void save_load_header(
         else  // < VERSION_FILE_WITH_INTERACTIONS
         {
           // pairs and triples may be restored but not reflected in interactions
-          all.interactions.insert(std::end(all.interactions), std::begin(all.pairs), std::end(all.pairs));
-          all.interactions.insert(std::end(all.interactions), std::begin(all.triples), std::end(all.triples));
+          std::vector<std::vector<namespace_index>> new_pairs(all.pairs.size());
```

Comment:
This doesn't reserve but creates N default elements. I think you meant to reserve here?